docs/user-guide/jobs.md

```diff
@@ -0,0 +1,256 @@
+
+In addition to required fields for a Pod, a pod template in a job must specify appropriate
+lables (see [pod selector](#pod-selector) and an appropriate restart policy.
+
+Only a [`RestartPolicy`](pod-states.md) equal to `Never` or `OnFailure` are allowed.
+
+### Pod Selector
+
+The `.spec.selector` field is a pod selector.  It works the same as the `.spec.selector` of
+a [ReplicationController](replication-controller.md).
+
+If specified, the `.spec.template.metadata.labels` must be equal to the `.spec.selector`.  Config not following
+this rule will be rejected by the API.  If unspecified, it will be defaulted to the latter.
+
+Also you should not normally create any pods whose labels match this selector, either directly,
+via another Job, or via another controller such as ReplicationController.  Otherwise, the Job will
+think that those pods were created by it.  Kubernetes will not stop you from doing this.
+
+### Multiple Completions
+
+By default, a Job is complete when one Pod runs to successful completion.  You can also specify that
+this needs to happen multiple times by specifying .spec.completions` with a value greater than 1.
+
+When multiple completions are requested, each Pod created by the Job controller has an identical
+[`spec`](../devel/api-conventions.md#spec-and-status).  In particular, all pods will have
+the same command line and the same image, the same volumes, and mostly the same environment
+variables.  It is up to the user to arrange for the pods to do work on different things.  For
+example, the pods might all access a shared work queue service to acquire work units.
+
+To create multiple pods which are similar, but have slightly different arguments, environment
+variables or images, use multiple Jobs.
+
+### Parallelism
+
+You can suggest how many pods should run concurrently by setting `.spec.parallelism` to the number
+of pods you would like to have running concurrently.  This number is a suggestion. The number
+running concurrently may be lower or higher for a variety of reasons.  For example, it may be lower
+if the number of remaining completions is less, or as the controller is ramping up, or if it is
+throttling the job due to excessive failures.  It may be higher for example if a pod is gracefully
+shutdown, and the replacement starts early.
+
+If you do not specify `.spec.parallelism`, it is up to the Jon controller to chose a value.
+
+## Handling Pod and Container Failures
+
+A Container in a Pod may fail for a number of reasons, such as because the process in it exited with
+a non-zero exit code, or the Container was killed for exceeding a memory limit, etc.  If this
+happens, and the `.spec.template.containers[].restartPolicy = "OnFailure"`, then the Pod stays
+on the node, but the Container is re-run.  Therefore, your program needs to handle the the case when it is
+restarted locally, or else specify `.spec.template.containers[].restartPolicy = "Never"`.
```

Comment:
Here again, I'm raising the issue of RestartPolicy, since my initial implementation differs from what we have right now, iow. now the RestartPolicy does not matter, we will always restart pods, which is not what the [API docs](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/experimental/types.go#L437-L439) says.